src/BenchmarkDotNet.Core/Toolchains/CsProj/CsProjGenerator.cs

```diff
@@ -24,21 +28,70 @@ protected override void GenerateProject(Benchmark benchmark, ArtifactsPaths arti
         {
             string template = ResourceHelper.LoadTemplate("CsProj.txt");
 
-            string content = SetPlatform(template, PlatformProvider(benchmark.Job.ResolveValue(EnvMode.PlatformCharacteristic, resolver)));
+            string platform = PlatformProvider(benchmark.Job.ResolveValue(EnvMode.PlatformCharacteristic, resolver));
+            string content = SetPlatform(template, platform);
             content = SetCodeFileName(content, Path.GetFileName(artifactsPaths.ProgramCodePath));
             content = SetDependencyToExecutingAssembly(content, benchmark.Target.Type);
             content = SetTargetFrameworkMoniker(content, TargetFrameworkMoniker);
-            //content = SetGcMode(content, benchmark.Job.Env.Gc, resolver); todo: implement
+            content = SetRuntimeIdentifier(content, platform);
+            content = content.Replace("$PROGRAMNAME$", artifactsPaths.ProgramName);
+
+            SetGcMode(content, benchmark.Job.Env.Gc, resolver, artifactsPaths);
 
             File.WriteAllText(artifactsPaths.ProjectFilePath, content);
         }
 
         protected override string SetDependencyToExecutingAssembly(string template, Type benchmarkTarget)
         {
+            if (!GetSolutionRootDirectory(out var solutionRootDirectory))
+            {
+                throw new NotSupportedException("Unable to find .sln or global.json file, hence can not find the csproj path");
+            }
+
             var assemblyName = benchmarkTarget.GetTypeInfo().Assembly.GetName();
+            var csprojName = $"{assemblyName.Name}.csproj";
+            var csprojs = solutionRootDirectory.GetFiles(csprojName, SearchOption.AllDirectories);
+            if (csprojs.Length != 1)
+            {
+                throw new NotSupportedException($"Unable to find single {csprojName} in {solutionRootDirectory} and it's subfolders. Most probably the name of output exe is different than the name of the .csproj");
+            }
+
+            return template.Replace("$CSPROJPATH$", csprojs.Single().FullName);
+        }
+
+        private void SetGcMode(string content, GcMode gcMode, IResolver resolver, ArtifactsPaths artifactsPaths)
+        {
+            if (!gcMode.HasChanges)
+                return;
+
+            var runtimeConfigContent = new StringBuilder(80)
```

Comment:
You don't have to generate this file anymore. See here: https://github.com/aspnet/KestrelHttpServer/pull/1362#pullrequestreview-21821949

cc @natemcmaster